src/System.Management.Automation/DscSupport/JsonCimDSCParser.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Security;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration
+{
+    /// <summary>
+    /// Class that does high level Cim schema parsing
+    /// </summary>
+    internal class CimDSCParser
+    {
+        private JsonDeserializer _json_deserializer;
+        
+        internal CimDSCParser()
+        {
+            _json_deserializer = JsonDeserializer.Create();
+        }
+
+        internal List<PSObject> ParseSchemaJson(string filePath)
+        {
+            string json = System.IO.File.ReadAllText(filePath);
+            try
+            {
+                string fileNameDefiningClass = Path.GetFileNameWithoutExtension(filePath);
+                int dotIndex = fileNameDefiningClass.IndexOf('.');
+                if (dotIndex != -1)
+                {
+                    fileNameDefiningClass = fileNameDefiningClass.Substring(0, dotIndex);
+                }
+
+                var result = new List<PSObject>(_json_deserializer.DeserializeClasses(json));
+                foreach (dynamic c in result)
+                {
+                    string superClassName = c.CimSuperClassName;
+                    string className = c.CimSystemProperties.ClassName;
+                    if ((superClassName != null) && (superClassName.Equals("OMI_BaseResource", StringComparison.OrdinalIgnoreCase)))
```

Comment:
Good point; updated.

Reply:
Because of `error CS0266: Cannot implicitly convert type 'bool?' to 'bool'. `
had to change it to 
`if (superClassName?.Equals("OMI_BaseResource", StringComparison.OrdinalIgnoreCase) ?? false)`